File: Local/Eye/src/main/java/br/com/eye/model/dao/StatementNotificacao.java
package br.com.eye.model.dao;

import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class StatementNotificacao {

    public static String getMensagemParaEnviarSlack(int codWorkspace) throws SQLException {
        int codNotificacao = 0;
        String mensagem = null;
        String sql = "SELECT cod_notificacao, mensagem FROM fila_notificacao WHERE cod_remetente = ? AND canal_envio = 1 AND remetente = 1 AND enviado = 0";
        PreparedStatement query = new Conexao().getConexao().prepareStatement(sql);
        query.setInt(1, codWorkspace);
        ResultSet resultado = query.executeQuery();
        if (resultado.next()) {
            codNotificacao = resultado.getInt("cod_notificacao");
            mensagem = resultado.getString("mensagem");
        }
        String update = "UPDATE fila_notificacao SET enviado = 1 WHERE cod_notificacao = ?";      
        PreparedStatement queryUpdate = new Conexao().getConexao().prepareStatement(update);
        queryUpdate.setInt(1, codNotificacao);
        queryUpdate.execute();
        return mensagem;
    }
    public static String getMensagemParaEnviarLocal (int codUsuario) throws SQLException {
        int codNotificacao = 0;
        String mensagem = null;
        String sql = "SELECT cod_notificacao, mensagem FROM fila_notificacao WHERE cod_remetente = ? AND canal_envio = 2 AND remetente = 2 AND enviado = 0";
        PreparedStatement query = new Conexao().getConexao().prepareStatement(sql);
        query.setInt(1, codUsuario);
        ResultSet resultado = query.executeQuery();
        if (resultado.next()) {
            codNotificacao = resultado.getInt("cod_notificacao");
            mensagem = resultado.getString("mensagem");
        }
        String update = "UPDATE fila_notificacao SET enviado = 1 WHERE cod_notificacao = ?";      
        PreparedStatement queryUpdate = new Conexao().getConexao().prepareStatement(update);
        queryUpdate.setInt(1, codNotificacao);
        queryUpdate.execute();
        return mensagem;
    }
}

File: Web/EYE/EYE/Model/DAO/StatementNotificacao.cs
﻿using EYE.Model.Enum;
using System.Data.SqlClient;

namespace EYE.Model.DAO
{
	public class StatementNotificacao
	{
		private const int MINIMO_DE_ALTERACAO = 1;
		public static bool RegistrarNotificacao(string mensagem, int codRemetente, CanalNotificacao canal, Remetente remetente)
		{
			using (var conexao = Conexao.GetConexao())
			{
				using (SqlCommand cmd = new SqlCommand("INSERT INTO fila_notificacao (mensagem, cod_remetente, canal_envio, remetente, enviado) VALUES (@mensagem, @cod_remetente, @canal_envio, @remetente, @enviado)", conexao))
				{
					cmd.Parameters.AddWithValue("@mensagem", mensagem);
					cmd.Parameters.AddWithValue("@cod_remetente", codRemetente);
					cmd.Parameters.AddWithValue("@canal_envio", canal);
					cmd.Parameters.AddWithValue("@remetente", remetente);
					cmd.Parameters.AddWithValue("@enviado", StatusVida.Desativado);
					return (cmd.ExecuteNonQuery() >= MINIMO_DE_ALTERACAO);
				}
			}
		}
	}
}
